backend/dataall/modules/metadata_forms/services/attached_metadata_form_service.py

```diff
@@ -35,13 +38,20 @@ class AttachedMetadataFormService:
     @staticmethod
     def create_attached_metadata_form(uri, data):
         AttachedMetadataFormValidationService.validate_filled_form_params(uri, data)
-        with get_context().db_engine.scoped_session() as session:
+        context = get_context()
+        with context.db_engine.scoped_session() as session:
             mf = MetadataFormRepository.get_metadata_form(session, uri)
             if not mf:
                 raise exceptions.ObjectNotFound('MetadataForm', uri)
             mf_fields = MetadataFormRepository.get_metadata_form_fields(session, uri)
             AttachedMetadataFormValidationService.validate_enrich_fields_params(mf_fields, data)
-
+            ResourcePolicyService.check_user_resource_permission(
+                session=session,
+                username=context.username,
+                groups=context.groups,
+                resource_uri=data.get('entityUri'),
+                permission_name=ATTACH_METADATA_FORM,
+            )
```

Comment:
any reason we don't use `    @ResourcePolicyService.has_resource_permission(ATTACH_METADATA_FORM, parent_resource=_FUNC_RETURNING_ENTITY_URI)` and instead explicitly call `check_user_resource_permission`

Reply:
because here we can't extract data.get('entityUri') as a parent resource.
We can, of course change the parameter of the function and pass 'entityUri' as a separate parameter, not as a part of data, but I assume it is more bulky, that just explicitly call `check_user_resource_permission`

Reply:
you can use `param_name` to resolve this, for example something similar to below:

```
    @ResourcePolicyService.has_resource_permission(
        ATTACH_METADATA_FORM, parent_resource=_get_entity_uri, param_name='data'
    )
    def create_attached_metadata_form(...):
```

where:
```
     def _get_entity_uri(data):
          reutnr data.get('entityUri')
```

`update_table_column_description` is an example where this `param_name` is used